src/surface.rs

```diff
@@ -0,0 +1,38 @@
+use std::rc::Rc;
+use std::cell::RefCell;
+use glium::backend::Facade;
+use glium::texture::{RawImage2d, Texture2d, Texture2dDataSource};
+
+use errors::*;
+
+/// A generic surface which we can render to.
+pub trait Surface {
+	/// Copies a buffer to the surface.
+    fn write_buffer(&mut self, buffer: &Vec<u8>, dimensions: (u32, u32)) -> Result<()>;
+}
+
+/// A `Surface` backed by a Glium `Texture2d`.
+pub struct OpenGLSurface {
+	pub texture: Texture2d,
```

Comment:
Probably should add a method to get a reference to this instead of using it directly